### PR TITLE
update and fix bug

### PR DESCRIPTION
**Pull request by Desplay**

### CODE REVIEW

File: src/auth/auth.resolver.ts [hunk 1]
```diff
@@ -1,54 +1,106 @@
...
+    const userTypeSignIn = {
+      NameOrEmail: user.username ? user.username : user.email,
+      password: user.password,
+    };
+    const authorization = await this.authService.SignIn(userTypeSignIn);
+    return authorization;
   }
 
-  @Mutation(() => JWT)
+  @Mutation(() => Authorization)
```
Comment (phanhuutoan) [inline src/auth/auth.resolver.ts:33]: AuthPayload !== Authorization

Comment (phanhuutoan) [reply src/auth/auth.resolver.ts:33]: Auth...Type

File: src/auth/auth.validate.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
```
Comment (phanhuutoan) [inline src/auth/auth.validate.ts:1]: authValidator.service.ts

File: src/auth/auth.validate.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    private readonly jwtService: JwtService,
+  ) {}
+
+  validateOTPCode(OTPCode: string, user: User): boolean {
+    if (user.OTPCode.verify) {
+      throw new ForbiddenException('OTP code is verified');
+    }
+    if (user.OTPCode.code !== OTPCode)
+      throw new ForbiddenException('Wrong OTP code');
+    if (user.OTPCode.dateExpired < new Date().getTime())
```
Comment (phanhuutoan) [inline src/auth/auth.validate.ts:22]: `{ OTPCode } = user`

Comment (phanhuutoan) [reply src/auth/auth.validate.ts:22]: throw new Exception('OTP code is invalid')

File: src/common/jwt/jwt.service.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import { Payload } from 'src/auth/datatype/payload.dto';
+import { UsersService } from 'src/users/users.service';
+@Injectable()
+export class JwtService {
+  constructor(
+    private readonly jwtService: JWTService,
+    private readonly usersService: UsersService,
+  ) {}
+
+  private validatePayload(payload: Payload): boolean {
```
Comment (phanhuutoan) [inline src/common/jwt/jwt.service.ts:12]: Best practice private xuống dưới

File: src/common/jwt/jwt.service.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    if (!payload) return false;
+    const Invalid_payload =
+      payload['user_id'] === undefined ||
+      payload['iat'] === undefined ||
+      payload['exp'] === undefined;
```
Comment (phanhuutoan) [inline src/common/jwt/jwt.service.ts:17]: `validPayload = payload[a] && payload[b] && payload[c]`

**Review by phanhuutoan — CHANGES_REQUESTED** [review]
Kiến trúc tương đối ổn, cần refactor thêm

**Comment by Desplay** [comment]
> Kiến trúc tương đối ổn, cần refactor thêm

all this fix done !

**Comment by Desplay** [comment]
> 

this is change: https://github.com/Desplay/NestJS_MongoDB_GraphQL_Basic/pull/5/commits/ffb24c9f9034c66153bcd568f44946fc658c59bc

**Comment by Desplay** [comment]
All in done, create Comments services & Like Services, allready merge in main brain

**Pull request merged**